src/util/amounts.ts

```diff
@@ -73,3 +74,24 @@ export function getApplicableV3FeeAmounts(chainId: ChainId): FeeAmount[] {
 
   return feeAmounts;
 }
+
+export function getApplicableV4FeesTickspacingsHooks(
+  chainId: ChainId
+): Array<[number, number, string]> {
+  const feeAmounts = [
+    FeeAmount.HIGH,
+    FeeAmount.MEDIUM,
+    FeeAmount.LOW,
+    FeeAmount.LOWEST,
+  ];
+
+  if (chainId === ChainId.BASE) {
+    feeAmounts.push(FeeAmount.LOW_200, FeeAmount.LOW_300, FeeAmount.LOW_400);
+  }
+
+  return feeAmounts.map((feeAmount) => [
+    feeAmount as number,
+    TICK_SPACINGS[feeAmount],
```

Comment:
this is v3 specific, for v4 it can be anything right? how are we planning on accounting for that?

Reply:
good question. I'm thinking:

1. from LP flow, I see the initial page encourage people to create v4 pools with the same fee-tiers (shown on the UI page) and the same corresponding tick spacings (not shown on the UI page). (This is still beta testing, so I don't wanna post screenshot here. Can DM you.) This means likely new v4 pools through web interface will have the same fee-tiers and tick spacings for majority of the v4 pools.
2. from SOR and routing-api, Im thinking to have SOR level default fee-tier and tick spacings the same as v3. Then at routing-api, I can pass in more fee-tier and tickspacings (and maybe hook address) combos as needed. The way I see this is that we only want to add more fee-tier and tickspacings in adhoc fashion, because otherwise fetching all potential fee-tier and tickspacings for v4 pools is not scalable during routing lambda runtime. There are 10k fee tiers (due to [solidity datatype](https://www.notion.so/Hookless-V4-Routing-8126b60c4b974b55ac5bc3dff4185581?d=d9d9d33ed0694b2ea2af5d3066938f14&pvs=4#ec641262184844cc915fb2f48cabb772)) and 32766 tick spacings ([tickspacing has granularity of two](https://uniswapteam.slack.com/archives/C07AD3507SQ/p1728587844162899), min 1, max 32767), hence 32 millions potential hookless v4 pools already. Routing has no idea which v4 pools out of which those are initialized.

Reply:
perhaps a 1-pager research on our options/limitations here can help us determine how to best approach this eventually.